support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -287,13 +302,77 @@ private Object parseValue(StaplerRequest request) throws ServletException, IOExc
     private Object convert(String name, ParameterValue v) throws IOException, InterruptedException {
         if (v instanceof FileParameterValue) {
             FileParameterValue fv = (FileParameterValue) v;
-            FilePath fp = new FilePath(run.getRootDir()).child(name);
-            fp.copyFrom(fv.getFile());
-            return fp;
+            FilePath workspace = getContext().get(FilePath.class);
+            if (workspace != null) {
+                FilePath fp =  workspace.child(name);
+                fp.copyFrom(fv.getFile());
+                return fp;
+            } else {
+                // This should never happen.
+                throw new Failure("No workspace found in context.");
+            }
         } else {
             return v.getValue();
         }
     }
 
     private static final long serialVersionUID = 1L;
+    
+    // TODO: replace with a call to MultipartFormDataParser.isMultipart once available in parent core
+    private static boolean isMultipart(StaplerRequest request) {
+        if (request == null) {
+            return false;
+        }
+
+        String contentType = request.getContentType();
+        if (contentType == null) {
+            return false;
+        }
+
+        String[] parts = contentType.split(";");
+        if (parts.length == 0) {
+            return false;
+        }
+
+        for (int i = 0; i < parts.length; i++) {
+            if ("multipart/form-data".equals(parts[i])) {
+                return true;
+            }
+        }
+
+        return false;
+    }    
+    
+    // TODO: total hack to gain access to all parsedFormData from Stapler RequestImpl
+    // It's a pity that RequestImpl makes the decision it does wrt isFormField FileItem filtering... why not let the 
+    // caller do that kind of filtering if they need it !!    
+    private static FileItem getFileItem(StaplerRequest request, String name) throws ServletException, IOException {
```

Comment:
I filed https://github.com/stapler/stapler/issues/52 to keep track of this task in stapler.